src/main/java/com/openshift/internal/restclient/apis/TypeMetaFactory.java

```diff
@@ -0,0 +1,106 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package com.openshift.internal.restclient.apis;
+
+import java.lang.reflect.Constructor;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.commons.lang.StringUtils;
+import org.jboss.dmr.ModelNode;
+
+import com.openshift.internal.restclient.api.models.TypeMeta;
+import com.openshift.internal.restclient.apis.autoscaling.models.Scale;
+import com.openshift.internal.restclient.model.properties.ResourcePropertiesRegistry;
+import com.openshift.internal.restclient.model.properties.ResourcePropertyKeys;
+import com.openshift.internal.util.JBossDmrExtentions;
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.ResourceFactoryException;
+import com.openshift.restclient.UnsupportedVersionException;
+import com.openshift.restclient.api.ITypeFactory;
+import com.openshift.restclient.api.models.INameSetable;
+import com.openshift.restclient.api.models.INamespaceSetable;
+import com.openshift.restclient.api.models.ITypeMeta;
+
+public class TypeMetaFactory implements ITypeFactory, ResourcePropertyKeys {
+	
+	private static final String DELIMITER = ".";
+	private static final Map<String, Class<? extends ITypeMeta>> IMPL_MAP = new HashMap<>();
+
+	static {
+		IMPL_MAP.put("Scale", Scale.class);
+	}
+	
+	@Override
+	public Object stubKind(String kind, Optional<String> name, Optional<String> namespace) {
```

Comment:
I like 'stub' better but I don't think I was able to reuse the method name because of the return type